python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1075,16 +1077,47 @@ class TimeWindowPartitionsSubset(PartitionsSubset):
     def __init__(
         self,
         partitions_def: TimeWindowPartitionsDefinition,
-        included_time_windows: Sequence[TimeWindow],
         num_partitions: int,
+        included_time_windows: Optional[Sequence[TimeWindow]] = None,
+        included_partition_keys: Optional[AbstractSet[str]] = None,
     ):
         self._partitions_def = check.inst_param(
             partitions_def, "partitions_def", TimeWindowPartitionsDefinition
         )
-        check.sequence_param(included_time_windows, "included_time_windows", of_type=TimeWindow)
         self._included_time_windows = included_time_windows
         self._num_partitions = num_partitions
 
+        check.param_invariant(
+            not (included_partition_keys and included_time_windows),
+            "Cannot specify both included_partition_keys and included_time_windows",
+        )
+        self._included_time_windows = check.opt_nullable_sequence_param(
+            included_time_windows, "included_time_windows", of_type=TimeWindow
+        )
+
+        pkeys = check.opt_nullable_set_param(
+            included_partition_keys, "included_partition_keys", of_type=str
+        )
+        self._included_partition_keys = (
+            sorted(
+                list(pkeys),
+                key=lambda pkey: self._partitions_def.start_time_for_partition_key(pkey),
```

Comment:
Is this not expensive? I don't actually know if this sorting matters.  When you asked me about the ordering of `get_partition_keys` on Slack, I thought you were asking about `PartitionsDefinition.get_partition_keys`.